File: Covenant.Api/Covenant.Common/Models/Worker/WorkerCommentModel.cs
namespace Covenant.Common.Models.Worker
{
    public class WorkerCommentModel
    {
        public Guid Id { get; set; }
        public string Comment { get; set; }
        public decimal Rate { get; set; }
        public string Logo { get; set; }
        public int NumberId { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: SigookApp/lib/features/comments/data/models/worker_comment_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../../domain/entities/worker_comment.dart';

part 'worker_comment_model.freezed.dart';
part 'worker_comment_model.g.dart';

@freezed
abstract class WorkerCommentModel with _$WorkerCommentModel {
  const WorkerCommentModel._();

  const factory WorkerCommentModel({
    required String id,
    required String comment,
    required double rate,
    String? logo,
    required int numberId,
    required DateTime createdAt,
  }) = _WorkerCommentModel;

  factory WorkerCommentModel.fromJson(Map<String, dynamic> json) =>
      _$WorkerCommentModelFromJson(json);

  WorkerComment toEntity() => WorkerComment(
        id: id,
        comment: comment,
        rate: rate,
        logo: logo,
        numberId: numberId,
        createdAt: createdAt,
      );
}
